### ISSUE TITLE
Minerva: Feedback Loop for Response Quality Improvement

### ISSUE THREAD

**Issue by pdenya** [mdxvision/mdx-vision-enterprise#129]
## Problem Description

AI systems improve through feedback, but currently Minerva has no mechanism for clinicians to provide explicit feedback on response quality. Without a feedback loop, there's no way to:
- Identify incorrect or suboptimal responses
- Learn which responses are helpful vs unhelpful
- Improve RAG retrieval accuracy
- Fine-tune prompt engineering
- Detect hallucinations or outdated information

A robust feedback system is essential for continuous quality improvement and clinician trust.

## Proposed Solution

Implement **multi-level feedback system** allowing clinicians to rate Minerva responses, report issues, and provide corrective input.

#### 1. **Inline Response Feedback** (Quick)
Voice and HUD-based quick feedback on individual responses:
- **Voice**: 
  - "Minerva, that was helpful" → thumbs up
  - "Minerva, that was wrong" → thumbs down
  - "Minerva, explain your reasoning" → show sources and logic
- **HUD**: Thumbs up/down icons on each Minerva response
- **Implicit**: Track whether clinician follows Minerva's suggestion (e.g., orders recommended test, documents suggested diagnosis)

#### 2. **Detailed Feedback Form** (In-depth)
For reporting specific issues:
```
Issue Type: 
  [ ] Incorrect information
  [ ] Outdated guideline
  [ ] Irrelevant response
  [ ] Missing context
  [ ] Hallucination (made up facts)
  [ ] Tone/style issue

Severity:
  [ ] Critical (patient safety concern)
  [ ] Important (workflow issue)
  [ ] Minor (preference/style)

Description: [Free text]

Corrected Information: [Optional - what should the response have been?]
```

#### 3. **Flagging for Review**
- "Minerva, flag that for review" → sends response + context to quality review queue
- Automatic flagging of low-confidence responses
- Flag critical domain responses (drug dosing, diagnoses) for expert review

#### 4. **Comparative Feedback**
- A/B test alternative responses
- "Here are two treatment options - which aligns better with your approach?"
- Learn from choices without explicit rating

### Feedback Storage and Processing

**Feedback Database Schema:**
```sql
CREATE TABLE minerva_feedback (
  id SERIAL PRIMARY KEY,
  clinician_id VARCHAR(50),
  session_id VARCHAR(50),
  response_id VARCHAR(50),
  timestamp TIMESTAMP,
  
  -- Feedback content
  feedback_type ENUM('thumbs_up', 'thumbs_down', 'flag', 'detailed_report'),
  severity ENUM('critical', 'important', 'minor'),
  issue_category ENUM('incorrect', 'outdated', 'irrelevant', 'hallucination', 'missing_context', 'tone'),
  
  -- Context
  user_query TEXT,
  minerva_response TEXT,
  rag_sources JSONB,  -- Which guidelines/docs were cited
  model_confidence FLOAT,
  
  -- Corrective input
  corrected_response TEXT,
  clinician_notes TEXT,
  
  -- Processing
  reviewed BOOLEAN DEFAULT FALSE,
  reviewer_id VARCHAR(50),
  action_taken TEXT,
  resolution_date TIMESTAMP
);
```

### Feedback Processing Workflow

1. **Immediate Actions** (Critical Feedback)
   - If `severity = 'critical'` → alert on-call AI team
   - Automatic email to MDx Vision clinical safety team
   - Temporarily disable similar responses until reviewed

2. **Daily Review** (Important/Minor Feedback)
   - Quality team reviews flagged responses
   - Update RAG knowledge base if guideline is outdated
   - Adjust prompt engineering if systematic issue detected
   - Add negative examples to few-shot prompts

3. **Weekly Analysis** (Aggregate Trends)
   - Identify most common feedback categories
   - Track thumbs up/down ratios over time (target >80% thumbs up)
   - Identify low-performing guideline sources
   - A/B test prompt variations based on feedback

4. **Monthly Fine-Tuning** (Long-term Improvement)
   - Use corrected responses as fine-tuning examples
   - Update retrieval weights for RAG
   - Retrain intent classifiers if systematic misunderstandings occur

### Feedback Visibility to Clinicians

**Personal Feedback Dashboard:**
- "My feedback history" - see all feedback submitted
- "Resolution status" - was my feedback addressed?
- "Impact" - how many other clinicians benefited from your feedback

**Aggregated Feedback Transparency:**
- Show thumbs up/down ratio on common queries: "92% of clinicians found this helpful"
- Display confidence scores: "I'm 85% confident in this response based on 3 guideline sources"

### Incentivization

Encourage high-quality feedback:
- **Leaderboard**: Top feedback contributors
- **Recognition**: "Your feedback improved Minerva for 500 clinicians this month"
- **CME Credits**: Structured feedback counts toward continuing education
- **Response Priority**: Active feedback providers get faster support responses

## Acceptance Criteria

- [ ] Implement voice commands: "Minerva, that was helpful/wrong/flag that"
- [ ] Add thumbs up/down UI buttons in HUD for each Minerva response
- [ ] Create detailed feedback form accessible via voice or HUD
- [ ] Implement feedback database with schema above
- [ ] Track implicit feedback (did clinician follow suggestion?)
- [ ] Store RAG sources with each response for feedback context
- [ ] Email alerts for critical severity feedback
- [ ] Quality review dashboard for MDx Vision team to process feedback
- [ ] Weekly automated feedback analysis report
- [ ] Display confidence scores on Minerva responses
- [ ] Show aggregated helpfulness ratings to clinicians
- [ ] Implement feedback resolution tracking (clinician notified when addressed)
- [ ] Export feedback data for model improvement and fine-tuning
- [ ] Privacy-preserving feedback (anonymize before sharing with model vendors)
- [ ] HIPAA compliance: scrub PHI from feedback before storage

## Technical Notes

- Store feedback in PostgreSQL with foreign keys to sessions and clinicians
- Use JSONB for flexible RAG source storage
- Implement webhook to Slack/email for critical feedback alerts
- Consider separate microservice for feedback processing (decoupled from real-time Minerva)
- Retention policy: Keep critical feedback indefinitely, minor feedback for 1 year
- Use feedback data to build "adversarial test sets" for regression testing

## Privacy & Security Considerations

- **PHI in Feedback**: User queries may contain PHI - anonymize before exporting for fine-tuning
- **Clinician Attribution**: Some clinicians may want anonymous feedback - support this
- **Transparency**: Clinicians should know how their feedback is used
- **Consent**: Explicit consent for using feedback in model training

## Related Features

- Feature #112 (Preference Learning) - feedback informs preference adaptation
- Feature #113 (Specialty-Specific) - specialty-specific feedback helps specialize responses
- Feature #115 (Custom Guidelines) - feedback helps identify when organization guidelines should be added

## Metrics for Success

- **Feedback Volume**: >20% of clinicians provide feedback monthly
- **Thumbs Up Ratio**: >80% of responses get thumbs up (or no thumbs down)
- **Issue Resolution Time**: Critical feedback addressed within 24 hours, important within 1 week
- **Response Quality Trend**: Thumbs up ratio increases over time (evidence of improvement)
- **Repeat Issues**: Same issue flagged <3 times (fix it properly the first time)

## Example Feedback Scenarios

**Scenario 1: Outdated Guideline**
- Clinician: "Minerva, what do you think about this hypertension?"
- Minerva: "Consider JNC-7 guideline threshold of 140/90"
- Clinician: "Minerva, that's outdated. ACC/AHA 2017 guidelines now use 130/80."
- System: Flags response, updates RAG knowledge base with ACC/AHA 2017, notifies clinician when fixed

**Scenario 2: Hallucination**
- Minerva: "Amoxicillin is contraindicated in patients with penicillin allergy and diabetes"
- Clinician: "Minerva, that's wrong - diabetes is not a contraindication."
- System: Critical feedback alert, response disabled, reviewed by clinical team, prompt adjusted

**Scenario 3: Preference Feedback**
- Minerva: "The patient presents with chest pain radiating to the left arm, associated with diaphoresis and nausea, concerning for acute coronary syndrome."
- Clinician: "Minerva, be more concise"
- System: Adjusts verbosity preference (Feature #112) and notes that long-form DDx responses get negative feedback

## References

- RLHF (Reinforcement Learning from Human Feedback) literature
- Clinical decision support feedback systems
- User feedback loop best practices
- HIPAA guidelines for de-identification of PHI in feedback data
